test/test_tensor_creation_ops.py

```diff
@@ -1960,6 +1960,54 @@ def test_inference(default_dtype):
         test_inference(torch.float64)
         test_inference(torch.float32)
 
+    @suppress_warnings
+    @onlyCPU
+    def test_tensor_factory_ndarrays(self, device):
+        """
+        This tests the case when the input to torch.tensor are multiple
+        numpy arrays that are embedded (recursively) within list(s). 
+        """
+
+        repeats = 10
+        np_types = [np.int8, np.int16, np.int32, np.int64, np.float16, np.float32,\
+                    np.float64, np.complex64, np.complex128]
+
+        for dtype in np_types: # Simple test
+            for _ in range(repeats):
+                arrays = [(10*np.random.randn(100)).astype(dtype) for _ in range(repeats)]
+                self.assertEqual(torch.tensor(arrays, device=device), torch.tensor(np.stack(arrays), device=device), exact_dtype=True, rtol=1e-05, atol=1e-08)
+
+                arrays = [(10*np.random.randn(10,10)).astype(dtype) for _ in range(repeats)]
+                self.assertEqual(torch.tensor(arrays, device=device), torch.tensor(np.stack(arrays), device=device), exact_dtype=True, rtol=1e-05, atol=1e-08)
+
+                arrays = [(10*np.random.randn(100,1)).astype(dtype) for _ in range(repeats)]
+                self.assertEqual(torch.tensor(arrays, device=device), torch.tensor(np.stack(arrays), device=device), exact_dtype=True, rtol=1e-05, atol=1e-08)
+
+
+        for dtype in np_types: # Simple test
```

Comment:
This test is very repetitious. Just create the arrays up front and up them in an iterable, then iterate over the product of them and the NumPy dtypes, cast the arrays to the NumPy dtype, and perform the test by comparing torch.tensor() to a reference (like from_numpy() or a custom reference).

Make sure there are empty arrays, and 0D-3D arrays in the test set, too. It's also great that you're looking at arrays which are not contiguous or views.